framework/material/FileSelectorDialog.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.studio.framework.material
+
+import androidx.compose.ui.awt.ComposeDialog
+import com.vaticle.typedb.studio.state.common.util.Property
+import java.awt.FileDialog
+import java.io.File
+import java.util.Optional
+import javax.swing.JFileChooser
+
+object FileSelectorDialog {
+    fun selectFilePath(parent: ComposeDialog, title: String, selectorOptions: SelectorOptions): String? {
+        val file = when (Property.OS.Current) {
+            Property.OS.MACOS -> macOSFileSelector(parent, title, selectorOptions)
+            else -> otherOSFileSelector(title, selectorOptions)
+        }
+
+        return file?.absolutePath
+    }
+
+    private fun macOSFileSelector(parent: ComposeDialog, title: String, selectorOptions: SelectorOptions): File? {
+        when (selectorOptions) {
+            SelectorOptions.FILES_ONLY -> System.setProperty("apple.awt.fileDialogForDirectories", "false")
+            SelectorOptions.DIRECTORIES_ONLY -> System.setProperty("apple.awt.fileDialogForDirectories", "true")
+        }
+
+        val fileDialog = FileDialog(parent, title, FileDialog.LOAD)
+        fileDialog.apply {
+            isMultipleMode = false
```

Comment:
We don't want users to select multiple files or folders in any instance of this being used.